services/scraper/src/server.ts

```
@@ -45,8 +49,17 @@ fastify.post('/scrape', async (request, reply) => {
 
-    const result = await scraper.scrape(new URLParser(validatedBody.data.url));
-    return result;
+    const { url, driver } = validatedBody.data;
+    const urlParser = new URLParser(url);
+
+    switch (driver) {
+      case 'puppeteer':
+        return await puppeteerScraper.scrape(urlParser);
+
+      default:
+        return await cheerioScraper.scrapeWithRetry(urlParser, 3, 1000);;
```

Biome (1.9.4)

[error] This code is unreachable

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)